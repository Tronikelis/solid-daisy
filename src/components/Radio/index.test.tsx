/* eslint-disable solid/reactivity */
import { render } from "@solidjs/testing-library";
import userEvent from "@testing-library/user-event";
import { createSignal } from "solid-js";
import { describe, expect, it, vi } from "vitest";

import { Radio, RadioItem } from ".";

describe("<Radio />", () => {
    it("renders correctly", () => {
        const { container } = render(() => (
            <Radio selected="" setSelected={vi.fn()}>
                <RadioItem label="what's up" />
            </Radio>
        ));

        expect(container).toMatchSnapshot();
    });

    it("controlled input works", async () => {
        const [selected, setSelected] = createSignal("");

        const screen = render(() => (
            <Radio selected={selected()} setSelected={setSelected}>
                <RadioItem data-testid="foo" value="foo" label="foo" />
                <RadioItem data-testid="bar" value="bar" label="bar" />
            </Radio>
        ));

        await userEvent.click(screen.getByText("foo"));

        expect(selected()).toBe("foo");
        expect(screen.getByTestId("foo")).toBeChecked();

        setSelected("bar");

        expect(screen.getByTestId("foo")).not.toBeChecked();
        expect(screen.getByTestId("bar")).toBeChecked();
    });
});
