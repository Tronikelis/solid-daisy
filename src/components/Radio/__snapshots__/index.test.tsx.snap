// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Radio /> > renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-0"
  >
    <div
      class="flex flex-row items-center gap-4"
    >
      <label
        class="label cursor-pointer gap-3"
      >
        <input
          class="radio"
          type="radio"
        />
        <span
          class="label-text"
        >
          what's up
        </span>
      </label>
    </div>
  </div>
</div>
`;
